### DISCUSSION TITLE
Is it possible to read for multiple tuple keys at a time?

### DISCUSSION THREAD

**Discussion by ragaeeb** [openfga/community#155] [Q&A]
👋 I would like to know if there is a way to read several tuple_keys in a single READ query.

Imagine the following scenario:
I have 4 objects: `a, b, c, d` which exist in a folder structure.

When the folder is deleted, I want to delete the tuples associated with object `a`, `b`, `c,` and `d`.

Right now when this happens, I have to do 4 separate READ requests like this:
```
{
   "tuple_key": {
       {
           "object": "a"
       }
   }
}
```

```
{
   "tuple_key": {
       {
           "object": "b"
       }
   }
}
```

etc.

Once I receive all these tuples back (after making at least 4 network requests), only then can I start issuing the delete requests for all of them.

If the folder has thousands of pages, then this will not scale well at all.

Is there a way to include `tuple_keys` as an array to pass in multiple objects at the same time?

**Comment by filip-stiglic** [comment]
Hi, any info on this? Because a batch endpoint would be great and definitely faster, than doing everyting in separate call.

**Comment by aaguiarz** [comment]
Hi @ragaeeb and @filipstiglic 

We are exploring adding more flexibility on Read, however, you can make those requests in parallel, and the impact in performance will be lower.

Additionally, you might also have the same data in your application's database (e.g. which documents are in a folder). If you do have it, you can just can delete the tuples without reading them.

@filipstiglic can you expand on your use case? Would you have the data in your application storage too?
